### PR TITLE
Add docs for themes extension type

### LINKED ISSUE #20555: Add docs for themes extension type
### Page

https://docs.directus.io/extensions/themes.html

### Describe the Inaccuracy

This page is missing from navigations, and also needs some work to add context.

### PR DESCRIPTION
**Pull request by phazonoverload**
This PR does three things:
1. Adds a themes section to the extensions overview page. 
2. Adds themes to the sidebar under developing extensions.
3. Cleans up the existing custom themes page created by @rijkvanzanten to provide additional context.

Originally I expected to also create a guide, but this particular extension type doesn't demand one as it can effectively be 'developed' in the Data Studio. One may follow, but is a nice-to-have.

Closes #20555

### CODE REVIEW

File: docs/extensions/themes.md [hunk 1]
```diff
@@ -26,42 +26,65 @@ export default defineTheme({
...
+	}
+}
+```
 
-https://github.com/directus/directus/blob/main/packages/themes/src/schemas/theme.ts
+Any rules that are not defined will fallback to the default theme for it's appearance. See the
+([`default dark theme`](https://github.com/directus/directus/blob/main/packages/themes/src/themes/dark/default.ts) and
+[`default light theme`](https://github.com/directus/directus/blob/main/packages/themes/src/themes/dark/default.ts)).
 
 We recommend using TypeScript for this extension type. The `defineTheme` function is typed to properly auto-complete all
```
Comment (paescuj) [inline docs/extensions/themes.md:54]:
```suggestion
We recommend using TypeScript for this extension type. The `defineTheme` function is typed to properly check and auto-complete all
```

File: docs/extensions/themes.md [hunk 2]
```diff
@@ -26,42 +26,65 @@ export default defineTheme({
 });
 ```
 
+### Available Options
+
+- `name` - The displayed name for this theme. This must be unique within your Directus project.
+- `appearance` - The section where this theme is displayed. Options are `'light'` or `'dark'`.
```
Comment (paescuj) [inline docs/extensions/themes.md:32]: I would put it more along these lines:
```suggestion
- `appearance` - To which appearance mode the theme belongs to, `light` or `dark`.
```

**Review by paescuj — APPROVED** [review]

**Review by paescuj — COMMENTED** [review]
Ah, can we add a sentence like this under "Google Fonts"?

> When using a Google Font, ensure the configured weight is available for the selected font.

**Review by rijkvanzanten — APPROVED** [review]

**Review by paescuj — APPROVED** [review]
Thanks ❤️

**Pull request merged**